package.json

```diff
@@ -1,7 +1,13 @@
 {
   "scripts": {
-    "compile-src": "rm -rf work/dist && tsc -p tsconfig-src.json && cp package.json work/dist",
-    "compile-test": "rm -rf work/dist-test && tsc -p tsconfig-test.json && cp package.json work/dist-test",
+    "precompile-src": "rm -rf work/dist",
+    "compile-src": "tsc -p tsconfig-src.json ",
+    "postcompile-src": "cp package.json work/dist",
+    "compile-test-ts": "tsc -p tsconfig-test.json",
+    "transpile-test-js": "BABEL_ENV=test babel work/dist-test --out-dir work/dist-test --source-maps",
```

Comment:
In order to get `nyc` to play with source maps properly, transpiling needed to be done _before_ tests are run so that source maps that `nyc` can use are generated.